### PR TITLE
Wrap and render newlines in JS dialogs

### LINKED ISSUE #4530: JS confirm/prompt doesn't wrap or handle newlines
**Version info (see `:version`)**:
recent
**Does the bug happen if you start with `--temp-basedir`?** (if applicable):
yes
**Description**
If a long string is passed to `window.confirm()` or `window.prompt()` anything that doesn't fit in one line of the prompt box isn't visible. If a multiline string is passed it will all be shown on one line.

On Google Chrome and in qutebrowser with `content.javascript.modal_dialog` set to true newlines are preserved and the message is wrapped to fit the dialog box.

I'm assuming that newlines aren't preserved because the dialog renders text as HTML which treats them as simple whitespace. The not wrapping thing appears to be because the QLabel isn't configured to do so.

This patch should do both of those things but I don't know if there are other ramifications. And I should finish my existing PRs before opening another:
```patch
diff --git i/qutebrowser/browser/shared.py w/qutebrowser/browser/shared.py
index 2398ca2e4d56..c08fc3fd45b0 100644
--- i/qutebrowser/browser/shared.py
+++ w/qutebrowser/browser/shared.py
@@ -82,6 +82,7 @@ def javascript_confirm(url, js_msg, abort_on, *, escape_msg=True):
         raise CallSuper
 
     js_msg = html.escape(js_msg) if escape_msg else js_msg
+    js_msg = js_msg.replace("\n", "<br/>")
     msg = 'From <b>{}</b>:<br/>{}'.format(html.escape(url.toDisplayString()),
                                           js_msg)
     urlstr = url.toString(QUrl.RemovePassword | QUrl.FullyEncoded)
diff --git i/qutebrowser/mainwindow/messageview.py w/qutebrowser/mainwindow/messageview.py
index 43ddd5248444..493fce71095a 100644
--- i/qutebrowser/mainwindow/messageview.py
+++ w/qutebrowser/mainwindow/messageview.py
@@ -35,6 +35,7 @@ class Message(QLabel):
         super().__init__(text, parent)
         self.replace = replace
         self.setAttribute(Qt.WA_StyledBackground, True)
+        self.setWordWrap(True)
         stylesheet = """
             padding-top: 2px;
             padding-bottom: 2px;

```
**How to reproduce**
On a page with javascript enabled open the web inspector console and past in
```javascript
window.confirm(`The best-known Japanese haiku is Bashō's "old pond":

old pond
frog leaps in
water's sound
`)
```
You can try with a [long string](https://en.wikipedia.org/wiki/Lorem_ipsum) yourself.

### PR DESCRIPTION
**Pull request by amacfie**
Closes #4530

### CODE REVIEW

File: qutebrowser/browser/shared.py [hunk 1]
```diff
@@ -79,14 +79,23 @@ def authentication_required(url, authenticator, abort_on):
     return answer
 
 
+def _format_msg(msg: str) -> str:
+    """Convert message to HTML suitable for rendering."""
+    ret = msg
```
Comment (The-Compiler) [inline qutebrowser/browser/shared.py:84]: Why not use `msg` directly below?

**Review by The-Compiler — CHANGES_REQUESTED** [review]

File: qutebrowser/browser/shared.py (hunk 1 above)
Comment (amacfie) [reply qutebrowser/browser/shared.py:84]: a matter of style i guess, so the lines that make changes can be rearranged freely. but i'm happy to change it

**Comment by The-Compiler** [comment]
Let me note that it'll take me a while until I get back to this (and other PRs) - I'm busy with giving a pytest course until tomorrow, then will focus on bugfixes for v2.0.2 and probably take a prolonged weekend off.

**Comment by The-Compiler** [comment]
Thanks! I removed the stripping in f8a2cad9aa054d16b899e6644f139467d4f8a8ca as I see no reason for it, and other browsers don't strip those messages either.

**Pull request merged**